pkg/services/ngalert/notifier/receivers.go

```diff
@@ -197,6 +199,39 @@ func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestRecei
 	return newTestReceiversResult(testAlert, append(invalid, results...), now), nil
 }
 
+func (am *Alertmanager) GetReceivers(ctx context.Context) apimodels.Receivers {
+	var receivers []*notify.Receiver
```

Comment:
Do we need this? Can't we just iterate over `am.receivers`?

Reply:
Do we need to copy the slice at all? There are no blocking operations here as far as I can tell can't we just hold the mutex for the duration of the function call?

Reply:
I thought this way we could release the lock as soon as possible to allow other goroutines to perform writes while we convert the copy of the slice to a format we could send back to the client.
Do you think deferring the call to `URnlock()` makes more sense in this case?

Reply:
I think it's nominal, we can hold the mutex for a 10-100us while we create the result.